doc/pages/user/041-astarte_device_sdks.md

```diff
@@ -0,0 +1,590 @@
+UUID v5 are obtained from a namespace UUID and a payload (a string).
+While all SDKs work with user-provided device ids, some also provide utilities to for UUID generation.
+
+C (ESP32) with an unique hardware ID using device MAC address and other identification bits:
+```c
+// deterministic id
+astarte_err_t astarte_hwid_get_id(&hw_id);
+```
+
+C++ (Qt5): not supported.
+
+Elixir: UUIDv5 can be obtained using the [elixir_uuid library](https://github.com/zyro/elixir-uuid).
+```elixir
+# random id
+device_id = :crypto.strong_rand_bytes(16) |> Base.url_encode64(padding: false)
+
+#deterministic id
+device_id = UUID.uuid5(namespace_uuid, payload, :raw)
+            |> Astarte.Core.Device.encode_device_id()
+```
+
+Go (using the [Astarte Go Client](https://github.com/astarte-platform/astarte-go)):
+```go 
+// Random id
+random_id, err := GenerateRandomAstarteId()
+
+// Namespaced id
+namespaced_id, err := GetNamespacedAstarteDeviceID(namespaceUuid,payload)
+```
+
+Java/Android: 
+```java
+// Namespaced id
+UUID deviceID = nameUUIDFromNamespaceAndString(namespaceUuid, payload);
+```
+
+Python: not supported.
+
+Rust:
+```rust
+/// Random id
+let random_uuid = astarte_sdk::registration::generate_random_uuid()
+
+///Namespaced id
+let namespaced_id = astarte_sdk::registration::generate_uuid(namespaceUuid, &payload) 
+```
+
+### Automatic Registration (Agent)
+You can refer to the [Astarte API for device registration](https://docs.astarte-platform.org/latest/api/index.html?urls.primaryName=Pairing%20API#/agent/registerDevice) for more details.
+
+C (ESP32):
+```c
+astarte_pairing_config cfg = 
+{
+    .base_url = &base_astarte_url;
+    .jwt = &jwt_token;
+    .realm = &realm;
+    .hw_id = &device_id;
+    .credentials_secret = &credentials_secret;
+};
+
+astarte_err_t err = astarte_pairing_register_device(&astarte_pairing_config);
+```
+
+C++ (Qt5): registration is done on device instantiation, see the next section.
+
+Elixir: 
+```elixir
+{:ok, %{body: %{"data" => %{"credentials_secret" => credentials_secret}}}} = Agent.register_device(client, device_id)
+```
+
+Go (using the [Astarte Go Client](https://github.com/astarte-platform/astarte-go)): 
+```go
+credentials_secret, err := client.Pairing.RegisterDevice(realm, deviceID) 
+```
+
+Java/Android: 
+```java
+AstartePairingService astartePairingService = new AstartePairingService(pairing_url, realm);
+String credentialsSecret = astartePairingService.registerDevice(jwt_token, device_id);
+```
+
+Python:
+```python
+credentials_secret = register_device_with_jwt_token(device_id, realm, jwt_token, pairing_base_url)
+```
+or 
+```python
+credentials_secret = register_device_with_private_key(device_id, realm, private_key_file, pairing_base_url)
+```
+
+Rust: 
+```rust
+    let credentials_secret =
+        astarte_sdk::registration::register_device(&jwt_token, &pairing_url, &realm, &device_id)
+            .await
+            .unwrap();
```

Comment:
I don't think it's really necessary to have the .unwrap() in the documentation, `.await?;` is sufficient for this purpose